src/ex3/pokemonClient.js

```diff
@@ -0,0 +1,134 @@
+import fetch from "node-fetch";
+import dotenv from "dotenv";
+
+import { pokemonTypeColor } from "./UI.js";
+import {
+  combineTwoArrays,
+  differenceOfTwoArray,
+  removeAllExceptNumbers,
+} from "./utils.js";
+
+dotenv.config();
+
+class PokemonClient {
+  pokemonApi = "https://pokeapi.co/api/v2/pokemon/";
+
+  handleFailedRequest(allRequestsResults) {
+    let failedIds;
+    let failedRquest;
+
+    failedRquest = allRequestsResults.filter((result) => {
+      return result.text.includes("not found");
+    });
+
+    if (failedRquest.length === 1) {
+      const text = failedRquest[0].text;
+      const id = removeAllExceptNumbers(text);
+      failedRquest = [{ id, text }];
+    } else if (failedRquest.length > 1) {
+      failedIds = failedRquest.map((obj) => {
+        return removeAllExceptNumbers(obj.text);
+      });
+
+      failedIds = failedIds.join(",");
+
+      const text = "Failed to fetch pokemon with this input: " + failedIds;
+      failedRquest = [
+        { id: failedIds, text, name: null, type: null, url: null },
+      ];
+    }
+
+    return failedRquest;
+  }
+  async fetchPokemon(id) {
+    const api = this.pokemonApi + id;
+    let text, result;
+    try {
+      const rawResponse = await fetch(api);
+      if (rawResponse.status === 404) {
+        text = "Pokemon with ID " + id + " was not found";
+        result = { id, text, name: null, type: null };
+      } else if (rawResponse.status === 200) {
+        const content = await rawResponse.json();
+        const url = content.sprites.other["official-artwork"].front_default;
+        const typesColored = this.getPokemonType(content.types)
+          .split("/")
+          .map((type) => {
+            return pokemonTypeColor[type](type);
+          })
+          .join("/");
+        text =
+          "Catch " + content.name + " the " + typesColored + " type pokemon";
+
+        result = {
+          id,
+          text,
+          name: content.name,
+          type: this.getPokemonType(content.types),
+        };
+      } else {
+        console.log("somthing wrong happend");
+        result = {
+          text: "Unknown error",
+        };
+      }
+    } catch (e) {
+      console.log(e);
+    }
+
+    return result;
+  }
+
+  async handleFetchNewPokemons(pokemonsIdToFetch) {
+    let allRequestsResults;
+    const allFetchRequest = pokemonsIdToFetch.map((pokemonId) => {
```

Comment:
allFetchRequest -> allFetchRequests
"all" is really good but "request" is confusing since its singular